hack/test_errata.py

```diff
@@ -0,0 +1,738 @@
+import datetime
+import errata
+import github
+import http
+import json
+import os
+import re
+import tempfile
+import unittest
+import urllib
+from collections import namedtuple
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+
+GithubUserMock = namedtuple("GithubUserMock", "login")
+GithubLabelMock = namedtuple("GithubLabelMock", "name")
+class GithubPRMock:
+    def __init__(self, user, title, labels=[], number=0, body="", url="", html_url=""):
+        self.user = user
+        self.title = title
+        self.labels = labels
+        self.number = number
+        self.body = body
+        self.url = url
+        self.html_url = html_url
+        self.create_issue_comment = MagicMock()
+        
+    def __eq__(self, other):
+        if not isinstance(other, GithubPRMock):
+            return False
+        return      self.user == other.user     \
+                and self.title == other.title   \
+                and self.labels == other.labels \
+                and self.number == other.number \
+                and self.body == other.body     \
+                and self.url == other.url       \
+                and self.html_url == other.html_url
+
+
+class ExtractErrataNumberFromBodyTest(unittest.TestCase):
+    def test_valid_url(self):
+        """
+        Test errata number extraction from valid URLs.
+        """
+        param_list = [
+            ('https://errata.devel.redhat.com/advisory/12345', 12345),
+            ('https://errata.devel.redhat.com/advisory/67890', 67890),
+            ('https://errata.devel.redhat.com/advisory/13579', 13579),
+            ('https://errata.devel.redhat.com/advisory/24680', 24680),
+            ('https://errata.devel.redhat.com/advisory/', None),
+            ('https://errata.devel.redhat.com/advisory/invalid', None)
+        ]
+        for (url, expected) in param_list:
+            with self.subTest():
```

Comment:
`subTest(url=url)` would [make it easier to identify the failing subtest][1], if we ever actually fail this.

[1]: https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest